field/character/classes/plugin.class.php

```diff
@@ -361,14 +361,16 @@ public function userform_mform_element($mform, $searchform, $readonly=false, $su
         $thresholdsize = 37;
         $lengthtochar = 1.3;
         $attributes = array('class' => 'indent-'.$this->indent, 'id' => $idprefix);
-        // Cool for browsers supporting html 5.
-        // if ($this->pattern == SURVEYPROFIELD_CHARACTER_EMAILPATTERN) {
-        //     $attributes['type'] = 'email';
-        // }
-        // if ($this->pattern == SURVEYPROFIELD_CHARACTER_URLPATTERN) {
-        //     $attributes['type'] = 'url';
-        // }
-        // But it doesn't work because "type" property is reserved to mform library
+        /*
```

Comment:
Not sure if code checker will like this alternative either. Anyway, np. Just commenting. Ciao :-)